permits/views.py

```diff
@@ -890,6 +891,14 @@ class PermitRequestList(SingleTableMixin, FilterView):
     paginate_by = int(os.environ["PAGINATE_BY"])
     template_name = "permits/permit_requests_list.html"
 
+    def render_to_response(self, context):
+        if (
+            services.is_2FA_mandatory(self.request.user)
+            and not self.request.user.is_verified()
+        ):
```

Comment:
I think this a wrong place for this check. It only enforces 2FA on this specific view. I think this condition&redirection should be done in a middleware or called on login (regardless the login's "next" view). Still looking at it.

Reply:
I think the best approach would be to extend the decorator: **otp_required**, so it checks the group configuration, and when needed calls its super() method. Then add right after every login_required decorators. WDYT?

Reply:
Moved the check to a decorator used after every login_required, works like a charm...